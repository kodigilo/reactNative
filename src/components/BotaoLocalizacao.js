import {ActivityIndicator, Button, PermissionsAndroid, Platform} from "react-native";
import React, {Component} from "react";
import Geolocation from "@react-native-community/geolocation";
import {setLoader, updateLocation} from './../actions'
import {fetchTempoAction} from '../actions/climaTempoAction'
import {fetchEnderecoAction} from '../actions/enderecoAction'
import {connect} from "react-redux";

class BotaoLocalizacao extends Component {

    constructor(props) {
        super(props)
        this.startLocation()
    }

    startLocation = () => {
        if (Platform.OS === 'ios') {
            this.getLocation();
        } else {
            const requestLocationPermission = async () => {
                const granted = await PermissionsAndroid.request(
                    PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
                    {
                        title: "Permissão de Acesso à Localização",
                        message: "Este aplicativo precisa acessar sua localização.",
                        buttonNeutral: "Pergunte-me depois",
                        buttonNegative: "Cancelar",
                        buttonPositive: "OK"
                    }
                );
                if (granted === PermissionsAndroid.RESULTS.GRANTED) {
                    this.getLocation();
                } else {
                    alert('Permissão de Localização negada');
                }
            };
            requestLocationPermission();
        }
    }

    getLocation = () => {

        this.props.dispatch(setLoader(true))

        Geolocation.getCurrentPosition(
            position => {
                this.props.dispatch(updateLocation({lat: position.coords.latitude, lon: position.coords.longitude}))
                this.props.dispatch(fetchTempoAction())
                this.props.dispatch(fetchEnderecoAction())
                this.props.dispatch(setLoader(false))
            },
            error => {
                Alert.alert('Error', JSON.stringify(error))
                this.props.dispatch(setLoader(false))
            },
            {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000},
        );

    }


    render() {
        return (
            <>
                <Button title={'Minha localização'} onPress={this.startLocation}/>
            </>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        location: state.location
    }
}
export default connect(mapStateToProps)(BotaoLocalizacao)