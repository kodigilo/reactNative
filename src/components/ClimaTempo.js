import {Text, View} from "react-native";
import React, {Component} from "react";
import {connect} from "react-redux";
import style from "../styles/climaTempoStyle"
import store from "../store";
import {fetchTempoAction} from "../actions/climaTempoAction";

class ClimaTempo extends Component {

    constructor(props) {
        super(props);
        store.dispatch(fetchTempoAction())
    }

    render() {
        return (
            <>
                {this.props.dados && Object.keys(this.props.dados).length > 0 &&
                    <View style={style.Relative}>
                        {this.props.dados.main && this.props.dados.main.temp && <Text style={style.Temp}>{parseInt(this.props.dados.main.temp)}º</Text>}
                        {this.props.dados.main && this.props.dados.main.temp_min && <View style={style.TempMin}><Text style={style.TempText}>{parseInt(this.props.dados.main.temp_min)}º</Text><Text style={style.TempLabel}>mín</Text></View>}
                        {this.props.dados.main && this.props.dados.main.temp_max && <View style={style.TempMax}><Text style={style.TempText}>{parseInt(this.props.dados.main.temp_max)}º</Text><Text style={style.TempLabel}>máx</Text></View>}
                        <View style={style.Row}>
                            {this.props.dados.sys && this.props.dados.sys.country && <Text style={style.Local}>{this.props.dados.name} - {this.props.dados.sys.country}</Text>}
                        </View>
                        {this.props.dados.main && this.props.dados.main.pressure && <Text><Text>Pressão: </Text> {this.props.dados.main.pressure}</Text>}
                        {this.props.dados.main && this.props.dados.main.humidity && <Text><Text>Humidade: </Text> {this.props.dados.main.humidity}</Text>}
                        {this.props.dados.name && !this.props.dados.sys && <Text><Text>Local: </Text> {this.props.dados.name}</Text>}
                        {this.props.dados.message && <Text><Text>Erro: </Text> {this.props.dados.message}</Text>}
                        <View style={style.Row}>
                            {Array.isArray(this.props.dados.weather) && this.props.dados.weather.map(item => {
                                return <Text style={style.Tags} key={item.id}>{item.description}</Text>
                            })}
                        </View>
                    </View>
                }
            </>
        )
    }

}

const mapStateToProps = (state) => {
    var tempo = state.tempoReduce.tempo
    if (tempo == undefined) tempo = {}
    return {dados: tempo}
}

export default connect(mapStateToProps)(ClimaTempo)