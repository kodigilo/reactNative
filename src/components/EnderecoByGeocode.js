import {Text,View} from "react-native";
import React, {Component, useState} from "react";
import {connect} from "react-redux";
import style from "../styles/enderecoStyle"
import store from "../store";
import {fetchEnderecoAction} from "../actions/enderecoAction";
class EnderecoByGeocode extends Component {

    constructor(props){
        super(props);
        store.dispatch(fetchEnderecoAction())
    }

    render() {
        return (
            <View style={style.Card}>
                {this.props.endereco && Object.keys(this.props.endereco).length > 0 &&
                    <>
                        <Text style={{fontSize: 20, fontWeight: 'bold'}}>Endere√ßo (aproximado):</Text>
                        {this.props.endereco.display_name && <Text>{this.props.endereco.display_name}</Text>}
                    </>
                }
            </View>
        )
    }
}

const mapStateToProps = (state) => {
    var endereco = state.enderecoReduce.endereco
    if (endereco == undefined) endereco = {}
    return {endereco: endereco}
}

export default connect(mapStateToProps)(EnderecoByGeocode)