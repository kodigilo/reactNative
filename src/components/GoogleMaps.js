import WebView from "react-native-webview";
import React, {Component} from "react";
import {LATITUDE, LONGITUDE, domain_google_maps} from '../../app.json';
import {connect} from 'react-redux'

class GoogleMaps extends Component {
    render() {
        return (<WebView source={{
            html: `<iframe src="${domain_google_maps}maps?q=${this.props.lat},${this.props.lon}&z=19&output=embed" width="100%" height="100%" style="border:0; " allowfullscreen="" loading="lazy"></iframe>`
        }}/>);
    }
}

const mapStateToProps = (state) => {
    const location = state.locationReduce.location
    if (location && Object.keys(location).length > 0 && location.lon)
        return location
    return {lat: LATITUDE, lon: LONGITUDE}
}

export default connect(mapStateToProps)(GoogleMaps)