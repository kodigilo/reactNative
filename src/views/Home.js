import React from "react";
import {ActivityIndicator, View} from "react-native";

import GoogleMaps from "../components/GoogleMaps";
import ClimaTempo from "../components/ClimaTempo";
import EnderecoByGeocode from "../components/EnderecoByGeocode";
import BotaoLocalizacao from "../components/BotaoLocalizacao";
import style from "../styles/homeStyle";
import {connect} from "react-redux";

const Home = (props) => {
    return (
        <>
            <View  style={[{
                flex: 1,
                flexDirection: "column"
            }]}>
                <View style={{flex: 2}}>
                    <GoogleMaps/>
                </View>
                <View style={{flex: 1, backgroundColor: '#fff', padding: 10}}>
                    <ClimaTempo/>
                    <EnderecoByGeocode/>
                    <BotaoLocalizacao/>
                </View>
            </View>
            { props.isActive && <ActivityIndicator size="large" color="#0000ff" style={style.loading}/> }
        </>
    );
};

const mapStateToProps = (state) => {
    return {
        isActive:  state.loaderReduce.loading
    }
}

export default connect(mapStateToProps)(Home)