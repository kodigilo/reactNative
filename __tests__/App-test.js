/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BotaoLocalizacao from "../src/components/BotaoLocalizacao";
import ClimaTempo from "../src/components/ClimaTempo";
import EnderecoByGeocode from "../src/components/EnderecoByGeocode";
import GoogleMaps from "../src/components/GoogleMaps";
import Home from "../src/views/Home";
import {Provider} from 'react-redux'
import store from '../src/store/'
import {fetch} from "react-native/Libraries/Network/fetch";

it('teste renderização botão localização', async () => {
    const tree = renderer.create(<Provider store={store}><BotaoLocalizacao/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('teste renderização infos clima tempo', async () => {
    const tree = renderer.create(<Provider store={store}><ClimaTempo/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('teste renderização infos endereço', async () => {
    const tree = renderer.create(<Provider store={store}><EnderecoByGeocode/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('teste renderização google maps', async () => {
    const tree = renderer.create(<Provider store={store}><GoogleMaps/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('teste renderização home', async () => {
    const tree = renderer.create(<Provider store={store}><Home/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});
