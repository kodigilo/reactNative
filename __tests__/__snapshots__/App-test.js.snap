// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste renderização botão localização 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Minha localização
    </Text>
  </View>
</View>
`;

exports[`teste renderização google maps 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "padding": 0,
      },
    ]
  }
>
  <RNCWebView
    cacheEnabled={true}
    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
    injectedJavaScriptForMainFrameOnly={true}
    javaScriptEnabled={true}
    messagingEnabled={false}
    onContentProcessDidTerminate={[Function]}
    onHttpError={[Function]}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    scalesPageToFit={false}
    source={
      Object {
        "html": "<iframe src=\\"http://maps.google.com/maps?q=-23.5613496,-46.665249&z=16&output=embed\\" width=\\"100%\\" height=\\"100%\\" style=\\"border:0; padding: 0; margin: 0; border: none \\" allowfullscreen=\\"\\" loading=\\"lazy\\"></iframe>",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    textInteractionEnabled={true}
    useSharedProcessPool={true}
  />
</View>
`;

exports[`teste renderização home 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "padding": 0,
            },
          ]
        }
      >
        <RNCWebView
          cacheEnabled={true}
          injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
          injectedJavaScriptForMainFrameOnly={true}
          javaScriptEnabled={true}
          messagingEnabled={false}
          onContentProcessDidTerminate={[Function]}
          onHttpError={[Function]}
          onLoadingError={[Function]}
          onLoadingFinish={[Function]}
          onLoadingProgress={[Function]}
          onLoadingStart={[Function]}
          onMessage={[Function]}
          onShouldStartLoadWithRequest={[Function]}
          scalesPageToFit={false}
          source={
            Object {
              "html": "<iframe src=\\"http://maps.google.com/maps?q=-23.5613496,-46.665249&z=16&output=embed\\" width=\\"100%\\" height=\\"100%\\" style=\\"border:0; padding: 0; margin: 0; border: none \\" allowfullscreen=\\"\\" loading=\\"lazy\\"></iframe>",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
            ]
          }
          textInteractionEnabled={true}
          useSharedProcessPool={true}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Minha localização
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <ActivityIndicator
    color="#0000ff"
    size="large"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0.5)",
        "border": 0,
        "bottom": 0,
        "elevation": 12,
        "justifyContent": "center",
        "left": 0,
        "margin": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
]
`;

exports[`teste renderização infos clima tempo 1`] = `null`;

exports[`teste renderização infos endereço 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
/>
`;

exports[`teste service 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Minha localização
    </Text>
  </View>
</View>
`;
